src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +32,156 @@ ActorInit Dm_Hina_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F63C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F75C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F9AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Draw.s")
+void DmHina_Init(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->unk180 = 1;
+    this->actionFunc = func_80A1F470;
+    this->unk_158 = this->actor.world.pos.y;
+    this->actor.focus.pos.y = this->actor.world.pos.y;
+    this->unk148 = 0.0f;
+    this->unk_15C = 1.0f;
+    this->actor.focus.pos.x = this->actor.world.pos.x;
+    this->actor.focus.pos.z = this->actor.world.pos.z;
+}
+
+void DmHina_Destroy(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+}
+
+void func_80A1F470(DmHina* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    Math_SmoothStepToF(&this->unk148, 0.6f, 0.5f, 0.05f, 0.001f);
+    this->unk154 = Math_SinS((s16)(play->gameplayFrames * 0x708)) * 8.0f;
+    if ((player->stateFlags1 & 0x400) && (this->actor.xzDistToPlayer < 80.0f)) {
+        this->unk180 = 0;
+        this->unk154 = 0.0f;
+        this->actor.world.pos.y += 40.0f;
+        this->actionFunc = func_80A1F56C;
+    }
+}
+
+void func_80A1F56C(DmHina* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == (u8)2) {
+        this->unk17C = 2;
+        this->actionFunc = func_80A1F5AC;
+    }
+}
+
+void func_80A1F5AC(DmHina* this, PlayState* play) {
+    this->unk17C--;
+    if (this->unk17C == 0) {
+        this->unk180 = 1;
+        Cutscene_StartManual(play, &play->csCtx);
+        this->unk17A = Play_CreateSubCamera(play);
+        Play_ChangeCameraStatus(play, 0, 1);
+        Play_ChangeCameraStatus(play, this->unk17A, 7);
+        this->actionFunc = func_80A1F63C;
+    }
+}
+
+void func_80A1F63C(DmHina* this, PlayState* play) {
+    this->unk160 = this->actor.world.pos.x + 100.0f;
+    this->unk164 = this->unk_158 + 260.0f;
+    this->unk168 = this->actor.world.pos.z + 100.0f;
+    this->unk16C = this->actor.world.pos.x;
+    this->unk174 = this->actor.world.pos.z;
+    this->unk170 = this->actor.world.pos.y + (this->unk154 * this->unk_15C) + (40.0f * this->unk_15C);
+    Play_SetCameraAtEye(play, this->unk17A, (Vec3f*)&this->unk16C, (Vec3f*)&this->unk160);
+    Math_SmoothStepToF(&this->actor.world.pos.y, this->unk_158 + 300.0f, 0.5f, 2.0f, 0.1f);
+    if (((this->unk_158 + 240.0f) < this->actor.world.pos.y) && (this->unk17E != 1)) {
+        this->unk17E = 1;
+        Actor_PlaySfx(&this->actor, 0x580BU);
+    }
+    Actor_PlaySfx(&this->actor, 0x219BU);
+}
+
+void func_80A1F75C(DmHina* this, PlayState* play) {
+    s32 var_a0;
+
+    switch (this->unk17E) {
+        case 0:
+            this->unk17F = 0;
+            this->unk14C = 0.0f;
+            break;
+        case 1:
```

Comment:
```suggestion
            break;

        case 1:
```
Newline please between cases (except fall-throughs). Helps tell them apart. Please do the same for the switch in the Draw function.